modules/prometheus/nixos-exporter.py

```diff
@@ -40,14 +58,20 @@ def get_version_id(self, path):
         return None
 
     def get_time(self, path):
-        # nix path-info --json /run/booted-system | jq .[0].registrationTime
+        # nix path-info --json /run/booted-system | jq .[].registrationTime
         result = subprocess.run(
             [ "nix", "path-info", "--json", path ],
             stdout=subprocess.PIPE
         )
         if result.returncode == 0:
             parsed = json.loads(result.stdout)
-            return parsed[0]['registrationTime']
+
+            # https://github.com/NixOS/nix/pull/9242
+            if self.nix_version >= "2.19.0":
```

Comment:
Will that work?
```
$ python3
>>> "2.3.16" >= "2.19.0"
True
```

Reply:
This one perhaps?
https://stackoverflow.com/a/11887825/587396

Reply:
I don't think we need wide version compatibility here, but the linked approach seems easy enough.

Reply:
Ah, yeah. That is annoying!

Reply:
Replaced the version check with a pretty standard library and tested both 2.18.1 and 2.19.0.